conans/test/functional/command/test_package_test.py

```diff
@@ -121,3 +122,40 @@ def build_folder_handling_test(self):
         self.assertTrue(os.path.exists(os.path.join(client.current_folder, "test_package",
                                                     "build_folder")))
         self.assertFalse(os.path.exists(default_build_dir))
+
+    def check_version_test(self):
+        client = TestClient()
+        dep = textwrap.dedent("""
+            from conans import ConanFile
+            class Dep(ConanFile):
+                def package_info(self):
+                    self.cpp_info.name = "MyDep"
+            """)
+        client.save({CONANFILE: dep})
+        client.run("create . dep/1.1@")
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile
+            class Pkg(ConanFile):
+                requires = "dep/1.1"
+                def build(self):
+                    info = self.deps_cpp_info["dep"]
+                    self.output.info("BUILD Dep %s VERSION %s" % (info.name, info.version))
+                def package_info(self):
+                    self.cpp_info.name = "MyHello"
+            """)
+        test_conanfile = textwrap.dedent("""
+            from conans import ConanFile
+            class Pkg(ConanFile):
+                def build(self):
+                    info = self.deps_cpp_info["hello"]
+                    self.output.info("BUILD HELLO %s VERSION %s" % (info.name, info.version))
+                def test(self):
+                    info = self.deps_cpp_info["hello"]
+                    self.output.info("TEST HELLO %s VERSION %s" % (info.name, info.version))
+            """)
+        client.save({"conanfile.py": conanfile,
+                     "test_package/conanfile.py": test_conanfile})
+        client.run("create . hello/0.1@")
+        self.assertIn("hello/0.1: BUILD Dep MyDep VERSION 1.1", client.out)
```

Comment:
I would expect `self.deps_cpp_info["xxxx"].name` to be the name of the package reference, not the value of `self.cpp_info.name`.

Reply:
This has NOT changed in this test, this is the way it was. I have just introduced the check to be consistent.

What I have fixed is the "Hello" one a few lines below. I am just dumping the ``cpp_info.name`` field in the ``conanbuildinfo.txt``.

Reply:
That's what I said. In this context (so close to the `version`) I would expect it to be the `name` part of the `reference`, while in another context I agree it should be `cpp_info.name` (or `cpp_info.names["generator"]`). I think it can be confusing and maybe it is worth to provide the actual reference:

```
assert self.deps_cpp_info["hello"].reference == "hello/0.1"
```

The user can parse it to get the name, the version or the user/channel.

----

The original issue was asking for a way to get the **`name` and the `version` of the package**, here we are providing just the `version`, where are we going to inform about the `name`?